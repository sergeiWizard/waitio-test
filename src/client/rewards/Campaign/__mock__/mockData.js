// eslint-disable-next-line import/prefer-default-export
export const initialState = {
  filterKey: 'all',
  history: {
    action: 'PUSH',
    length: 46,
  },
  proposition: {
    count: 1,
    distance: null,
    last_created: '2020-02-04T12:32:15.446Z',
    max_reward: 0.001,
    min_reward: 0.001,
  },
};
