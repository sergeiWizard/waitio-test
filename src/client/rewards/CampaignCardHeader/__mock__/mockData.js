// eslint-disable-next-line import/prefer-default-export
export const initialState = {
  campaignData: {
    __v: 0,
    _id: '5e5e6a33fb97af17249e3f7f',
    activation_permlink: 'activate-monterey-7y5q4n886t5',
    app: 'waiviodev',
    budget: 1,
    commissionAgreement: 0.05,
    compensationAccount: '',
    count_objects: 1,
    count_reservation_days: 7,
    count_users: 0,
    createdAt: '2020-03-03T14:31:15.970Z',
    expired_at: '2020-03-07T14:31:11.877Z',
    frequency_assign: 0,
    frequency_exceed: false,
    guide: {
      alias: 'Alex Grigurko',
      name: 'alexeygrigurko',
      totalPayed: 1.819,
      wobjects_weight: 1496742.3025625013,
    },
    is_sponsor_blacklist: false,
    max_assign_count: 1,
    name: 'Testing123',
    requiredObject: 'Testing123',
    reward: 0.001,
    status: 'active',
    type: 'reviews',
    updatedAt: '2020-03-03T14:31:25.908Z',
    usersLegalNotice: '',
  },
  isDetails: false,
};

export const stateForModalDetail = {
  campaignData: {
    __v: 0,
    _id: '5e5e6a33fb97af17249e3f7f',
    activation_permlink: 'activate-monterey-7y5q4n886t5',
    app: 'waiviodev',
    budget: 1,
    commissionAgreement: 0.05,
    compensationAccount: '',
    count_objects: 1,
    count_reservation_days: 7,
    count_users: 0,
    createdAt: '2020-03-03T14:31:15.970Z',
    expired_at: '2020-03-07T14:31:11.877Z',
    frequency_assign: 0,
    frequency_exceed: false,
    guide: {
      alias: 'Alex Grigurko',
      name: 'alexeygrigurko',
      totalPayed: 1.819,
      wobjects_weight: 1496742.3025625013,
    },
    is_sponsor_blacklist: false,
    max_assign_count: 1,
    name: 'Testing123',
    requiredObject: 'Testing123',
    reward: 0.001,
    status: 'active',
    type: 'reviews',
    updatedAt: '2020-03-03T14:31:25.908Z',
    usersLegalNotice: '',
  },
  isDetails: true,
};

export const stateForTotal = {
  campaignData: {
    __v: 0,
    _id: '5e5e6a33fb97af17249e3f7f',
    activation_permlink: 'activate-monterey-7y5q4n886t5',
    app: 'waiviodev',
    budget: 1,
    commissionAgreement: 0.05,
    compensationAccount: '',
    count_objects: 1,
    count_reservation_days: 7,
    count_users: 0,
    createdAt: '2020-03-03T14:31:15.970Z',
    expired_at: '2020-03-07T14:31:11.877Z',
    frequency_assign: 0,
    frequency_exceed: false,
    guide: {
      alias: 'Alex Grigurko',
      name: 'alexeygrigurko',
      totalPayed: 0,
      wobjects_weight: 1496742.3025625013,
    },
    is_sponsor_blacklist: false,
    max_assign_count: 1,
    name: 'Testing123',
    requiredObject: 'Testing123',
    reward: 0.001,
    status: 'active',
    type: 'reviews',
    updatedAt: '2020-03-03T14:31:25.908Z',
    usersLegalNotice: '',
  },
  isDetails: true,
};

export const stateForModalSteem = {
  campaignData: {
    __v: 0,
    _id: '5e5e6a33fb97af17249e3f7f',
    activation_permlink: 'activate-monterey-7y5q4n886t5',
    app: 'waiviodev',
    budget: 1,
    commissionAgreement: 0.05,
    compensationAccount: '',
    count_objects: 1,
    count_reservation_days: 7,
    count_users: 0,
    createdAt: '2020-03-03T14:31:15.970Z',
    expired_at: '2020-03-07T14:31:11.877Z',
    frequency_assign: 0,
    frequency_exceed: false,
    guide: {
      alias: 'Alex Grigurko',
      name: 'alexeygrigurko',
      totalPayed: 0,
      wobjects_weight: 1496742.3025625013,
    },
    is_sponsor_blacklist: false,
    max_assign_count: 1,
    name: 'Testing123',
    requiredObject: 'Testing123',
    reward: 0,
    status: 'active',
    type: 'reviews',
    updatedAt: '2020-03-03T14:31:25.908Z',
    usersLegalNotice: '',
  },
  isDetails: true,
};

export const stateForModalSteemMinus = {
  campaignData: {
    __v: 0,
    _id: '5e5e6a33fb97af17249e3f7f',
    activation_permlink: 'activate-monterey-7y5q4n886t5',
    app: 'waiviodev',
    budget: 1,
    commissionAgreement: 0.05,
    compensationAccount: '',
    count_objects: 1,
    count_reservation_days: 7,
    count_users: 0,
    createdAt: '2020-03-03T14:31:15.970Z',
    expired_at: '2020-03-07T14:31:11.877Z',
    frequency_assign: 0,
    frequency_exceed: false,
    guide: {
      alias: 'Alex Grigurko',
      name: 'alexeygrigurko',
      totalPayed: 0,
      wobjects_weight: 1496742.3025625013,
    },
    is_sponsor_blacklist: false,
    max_assign_count: 1,
    name: 'Testing123',
    requiredObject: 'Testing123',
    reward: -1,
    status: 'active',
    type: 'reviews',
    updatedAt: '2020-03-03T14:31:25.908Z',
    usersLegalNotice: '',
  },
  isDetails: true,
};
