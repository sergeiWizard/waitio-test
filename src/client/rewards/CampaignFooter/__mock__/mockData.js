// eslint-disable-next-line import/prefer-default-export
export const initialState = {
  ownPost: false,
  pendingLike: false,
  pendingFlag: false,
  pendingFollow: false,
  pendingFollowObject: false,
  pendingBookmark: false,
  saving: false,
  onLikeClick: () => {},
  onCommentClick: () => {},
  handlePostPopoverMenuClick: () => {},
  toggleModalDetails: () => {},
};
