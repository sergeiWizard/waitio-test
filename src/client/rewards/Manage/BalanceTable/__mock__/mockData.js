export const initialState = {
  budgetTotal: {
    remaining: 6.95,
    sum_budget: 6.95,
    sum_payable: 0,
    sum_reserved: 0,
  },
  user: {
    active: {
      account_auths: [],
      key_auths: ['STM8MKLK4gtD8d2vJbzUZwvtoasxCziMiiPQFjQH8cXfr6YdKkhSF', '1'],
      weight_threshold: 1,
    },
    balance: '128.710 STEEM',
    can_vote: true,
    comment_count: 0,
    created: '2019-04-23T07:49:21',
    curation_rewards: 26127,
    delegated_vesting_shares: '0.000000 VESTS',
    downvote_manabar: {
      current_mana: '502592455076',
      last_update_time: '1583490708',
    },
    guest_bloggers: [],
    id: 1262161,
    last_account_recovery: '1970-01-01T00:00:00',
    last_account_update: '"2020-02-10T15:14:30"',
    last_owner_update: '1970-01-01T00:00:00',
    last_post: '2020-03-06T11:07:09',
    last_root_post: '2020-02-25T13:13:57',
    last_vote_time: '2020-03-06T10:31:48',
    lifetime_vote_count: 0,
    market_history: [],
    memo_key: 'STM8a1puEKRPxJRtpUeZcTEK1NyofCYNVaiNXCKPBTd3pBZn3jVcu',
    mined: false,
    name: 'asd09',
    next_vesting_withdrawal: '1969-12-31T23:59:59',
    other_history: [],
    owner: {
      account_auths: [],
      key_auths: ['STM84W9pJmUjdR3D2QaEhu1RwBvxxGcd4VeWBSq1hAVRs7ym3sZGe', 1],
      weight_threshold: 1,
    },
    pending_claimed_accounts: 0,
    post_bandwidth: 0,
    post_count: 40397,
    post_history: [],
    posting: {
      account_auths: [
        ['busy.app', 1],
        ['investarena', 1],
        ['jucci', 1],
      ],
      key_auths: ['STM6nEnuY3jRSCGB5F4cvBefRjDxDN8ProZ3166KKQCUUgFmcHD3p', 1],
      weight_threshold: 1,
    },
    posting_rewards: 678,
    proxied_vsf_votes: [0, 0, 0, 0],
    proxy: '',
    received_vesting_shares: '1997152.344711 VESTS',
    recovery_account: 'blocktrades',
    reputation: -1792344993797,
    reset_account: 'null',
    reward_sbd_balance: '0.000 SBD',
    reward_steem_balance: '0.000 STEEM',
    reward_vesting_balance: '0.000000 VESTS',
    reward_vesting_steem: '0.000 STEEM',
    savings_balance: '0.000 STEEM',
    savings_sbd_balance: '0.000 SBD',
    savings_sbd_last_interest_payment: '1970-01-01T00:00:00',
    savings_sbd_seconds: '0',
    savings_sbd_seconds_last_update: '1970-01-01T00:00:00',
    savings_withdraw_requests: 0,
    sbd_balance: '1.000 SBD',
    sbd_last_interest_payment: '2020-02-26T00:00:48',
    sbd_seconds: '777591000',
    sbd_seconds_last_update: '2020-03-06T00:00:39',
    tags_usage: [],
    to_withdraw: '265008298803',
    transfer_history: [],
    vesting_balance: '0.000 STEEM',
    vesting_shares: '13217.475594 VESTS',
    vesting_withdraw_rate: '0.000000 VESTS',
    vote_history: [],
    voting_manabar: {
      current_mana: '1765315326783',
      last_update_time: 1583494296,
    },
    voting_power: 8781,
    withdraw_routes: 0,
    withdrawn: '265008298803',
    witness_votes: [],
    witnesses_voted_for: 0,
  },
};

export const stateToOperations = {
  budgetTotal: {
    remaining: 6.95,
    sum_budget: 6.95,
    sum_payable: 50,
    sum_reserved: 50,
  },
  user: {
    active: {
      account_auths: [],
      key_auths: ['STM8MKLK4gtD8d2vJbzUZwvtoasxCziMiiPQFjQH8cXfr6YdKkhSF', '1'],
      weight_threshold: 1,
    },
    balance: '128.710 STEEM',
    can_vote: true,
    comment_count: 0,
    created: '2019-04-23T07:49:21',
    curation_rewards: 26127,
    delegated_vesting_shares: '0.000000 VESTS',
    downvote_manabar: {
      current_mana: '502592455076',
      last_update_time: '1583490708',
    },
    guest_bloggers: [],
    id: 1262161,
    last_account_recovery: '1970-01-01T00:00:00',
    last_account_update: '"2020-02-10T15:14:30"',
    last_owner_update: '1970-01-01T00:00:00',
    last_post: '2020-03-06T11:07:09',
    last_root_post: '2020-02-25T13:13:57',
    last_vote_time: '2020-03-06T10:31:48',
    lifetime_vote_count: 0,
    market_history: [],
    memo_key: 'STM8a1puEKRPxJRtpUeZcTEK1NyofCYNVaiNXCKPBTd3pBZn3jVcu',
    mined: false,
    name: 'asd09',
    next_vesting_withdrawal: '1969-12-31T23:59:59',
    other_history: [],
    owner: {
      account_auths: [],
      key_auths: ['STM84W9pJmUjdR3D2QaEhu1RwBvxxGcd4VeWBSq1hAVRs7ym3sZGe', 1],
      weight_threshold: 1,
    },
    pending_claimed_accounts: 0,
    post_bandwidth: 0,
    post_count: 40397,
    post_history: [],
    posting: {
      account_auths: [
        ['busy.app', 1],
        ['investarena', 1],
        ['jucci', 1],
      ],
      key_auths: ['STM6nEnuY3jRSCGB5F4cvBefRjDxDN8ProZ3166KKQCUUgFmcHD3p', 1],
      weight_threshold: 1,
    },
    posting_rewards: 678,
    proxied_vsf_votes: [0, 0, 0, 0],
    proxy: '',
    received_vesting_shares: '1997152.344711 VESTS',
    recovery_account: 'blocktrades',
    reputation: -1792344993797,
    reset_account: 'null',
    reward_sbd_balance: '0.000 SBD',
    reward_steem_balance: '0.000 STEEM',
    reward_vesting_balance: '0.000000 VESTS',
    reward_vesting_steem: '0.000 STEEM',
    savings_balance: '0.000 STEEM',
    savings_sbd_balance: '0.000 SBD',
    savings_sbd_last_interest_payment: '1970-01-01T00:00:00',
    savings_sbd_seconds: '0',
    savings_sbd_seconds_last_update: '1970-01-01T00:00:00',
    savings_withdraw_requests: 0,
    sbd_balance: '1.000 SBD',
    sbd_last_interest_payment: '2020-02-26T00:00:48',
    sbd_seconds: '777591000',
    sbd_seconds_last_update: '2020-03-06T00:00:39',
    tags_usage: [],
    to_withdraw: '265008298803',
    transfer_history: [],
    vesting_balance: '0.000 STEEM',
    vesting_shares: '13217.475594 VESTS',
    vesting_withdraw_rate: '0.000000 VESTS',
    vote_history: [],
    voting_manabar: {
      current_mana: '1765315326783',
      last_update_time: 1583494296,
    },
    voting_power: 8781,
    withdraw_routes: 0,
    withdrawn: '265008298803',
    witness_votes: [],
    witnesses_voted_for: 0,
  },
};
