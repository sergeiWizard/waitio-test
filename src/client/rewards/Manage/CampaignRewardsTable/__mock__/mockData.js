// eslint-disable-next-line import/prefer-default-export
export const initialState = {
  campaigns: [
    {
      _id: '5e620da2fb97af17249e3f95',
      activation_permlink: 'activate-monterey-ln7osqo19e',
      budget: 0.1,
      completed: 0,
      name: 'TstCampa',
      remaining: 1,
      reserved: 0,
      reward: 0.1,
      status: 'active',
      type: 'reviews',
    },
    {
      _id: '5e38165d31b1bb6e49b265b6',
      activation_permlink: 'activate-monterey-4znrq130xhe',
      budget: 0.2,
      completed: 0,
      name: 'capa',
      remaining: 0,
      reserved: 0,
      reward: 0.1,
      status: 'expired',
      type: 'reviews',
    },
    {
      _id: '5e3815b431b1bb6e49b265b4',
      activation_permlink: 'activate-monterey-4p0t3jnbpmr',
      budget: 0.1,
      completed: 0,
      name: 'capaz1',
      remaining: 0,
      reserved: 0,
      reward: 0.1,
      status: 'expired',
      type: 'reviews',
    },
  ],
  userName: 'asd09',
};
