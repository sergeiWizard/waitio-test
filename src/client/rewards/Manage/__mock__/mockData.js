// eslint-disable-next-line import/prefer-default-export
export const initialState = {
  userName: 'asd09',
  user: {
    active: {},
    balance: '118.710 STEEM',
    can_vote: true,
    name: 'asd09',
  },
};

export const initialStateToDidMount = {
  campaigns: [],
  budgetTotal: {},
};

export const stateToComponentDidMount = {
  budgetTotal: {
    remaining: 6.95,
    sum_budget: 6.96,
    sum_payable: 0.01,
    sum_reserved: 0,
  },
  campaigns: [
    {
      _id: '5e678a1ff9c2cb174722106c',
      activation_permlink: 'activate-monterey-hkvru6omohm',
      budget: 0.01,
      completed: 1,
      name: 'CapaZeka',
      remaining: 0,
      reserved: 0,
      reward: 0.01,
      status: 'active',
      type: 'reviews',
    },
    {
      id: '5e620da2fb97af17249e3f95',
      activation_permlink: 'activate-monterey-ln7osqo19e',
      budget: 0.01,
      completed: 0,
      name: 'TstCampa',
      remaining: 0,
      reserved: 0,
      reward: 0.1,
      status: 'expired',
      type: 'reviews',
    },
  ],
};
