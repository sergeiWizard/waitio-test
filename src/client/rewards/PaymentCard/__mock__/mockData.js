export const initialState = {
  alias: 'Waivio Service',
  history: {
    action: 'POP',
    block: 'func',
    createHref: 'func',
    go: 'func',
    goBack: 'func',
    goForward: 'func',
    length: 50,
    listen: 'func',
    location: {
      hash: '',
      key: 'xq6l95',
      pathname: '/rewards/payables',
      search: '',
      state: null,
    },
  },
  location: {
    hash: '',
    key: 'xq6l95',
    pathname: '/rewards/payables',
    search: '',
    state: null,
  },
  match: {
    isExact: true,
    params: {},
    path: '/rewards/payables',
    url: '/rewards/payables',
  },
  name: 'wiv01',
  path: '/rewards/payables/@wiv01',
  payable: 0.01,
  staticContext: undefined,
};

export const stateForMemo = {
  alias: 'Waivio Service',
  history: {
    action: 'POP',
    block: 'func',
    createHref: 'func',
    go: 'func',
    goBack: 'func',
    goForward: 'func',
    length: 50,
    listen: 'func',
    location: {
      hash: '',
      key: 'xq6l95',
      pathname: '/rewards/payables',
      search: '',
      state: null,
    },
    push: 'func',
    replace: 'func',
  },
  location: {
    hash: '',
    key: 'xq6l95',
    pathname: '/rewards/payables',
    search: '',
    state: null,
  },
  match: {
    isExact: true,
    params: {},
    path: '/rewards/payables',
    url: '/rewards/payables',
  },
  name: 'waivio_',
  path: '/rewards/payables/@wiv01',
  payable: 0.01,
  staticContext: undefined,
};

export const stateForChangeMatchPath = {
  alias: 'Waivio Service',
  history: {
    action: 'POP',
    block: 'func',
    createHref: 'func',
    go: 'func',
    goBack: 'func',
    goForward: 'func',
    length: 50,
    listen: 'func',
    location: {
      hash: '',
      key: 'xq6l95',
      pathname: '/rewards/receivables',
      search: '',
      state: null,
    },
    push: 'func',
    replace: 'func',
  },
  location: {
    hash: '',
    key: 'xq6l95',
    pathname: '/rewards/receivables',
    search: '',
    state: null,
  },
  match: {
    isExact: true,
    params: {},
    path: '/rewards/receivables',
    url: '/rewards/receivables',
  },
  name: 'waivio_',
  path: '/rewards/receivables/',
  payable: 0.01,
  staticContext: undefined,
};
