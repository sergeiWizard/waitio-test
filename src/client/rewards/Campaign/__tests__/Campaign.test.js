import React from 'react';
// import React, { useContext } from 'react';
import { shallow } from 'enzyme';
import { mountWithIntl } from 'enzyme-react-intl';
import { MemoryRouter as Router } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { IntlProvider } from 'react-intl';
import Campaign from '../Campaign';
import { initialState } from '../__mock__/mockData';

// jest.mock('../Campaign', () => () => <div className="context-wrap" />);
jest.mock('../../../Wrapper', () => () => <div className="context-wrap" />);

describe('Campaign snapshot', () => {
  jest.mock('../Campaign', () => () => <div className="Campaign-mock" />);
  let wrapper;

  it('renders and matches snapshot', () => {
    const props = initialState;
    wrapper = shallow(
      <IntlProvider>
        <Campaign {...props} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Campaign', () => {
  let wrapper;

  beforeEach(() => {
    const props = initialState;

    wrapper = mountWithIntl(
      <Router>
        <Campaign {...props} />
      </Router>,
    );
  });

  afterEach(() => jest.clearAllMocks());

  it('should render Campaign wrap', () => {
    act(() => {
      wrapper.update();
    });
    const container = wrapper.find('.Campaign');
    expect(container).toHaveLength(2);
  });
});
