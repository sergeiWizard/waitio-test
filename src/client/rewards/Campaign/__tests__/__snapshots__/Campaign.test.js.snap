// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign aa renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <InjectIntl(withRouter(Campaign))
      filterKey="all"
      history={
        Object {
          "action": "PUSH",
          "length": 46,
        }
      }
      proposition={
        Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        }
      }
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "filterKey": "all",
      "history": Object {
        "action": "PUSH",
        "length": 46,
      },
      "proposition": Object {
        "count": 1,
        "distance": null,
        "last_created": "2020-02-04T12:32:15.446Z",
        "max_reward": 0.001,
        "min_reward": 0.001,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "filterKey": "all",
        "history": Object {
          "action": "PUSH",
          "length": 46,
        },
        "proposition": Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;

exports[`Campaign renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <InjectIntl(withRouter(Campaign))
      filterKey="all"
      history={
        Object {
          "action": "PUSH",
          "length": 46,
        }
      }
      proposition={
        Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        }
      }
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "filterKey": "all",
      "history": Object {
        "action": "PUSH",
        "length": 46,
      },
      "proposition": Object {
        "count": 1,
        "distance": null,
        "last_created": "2020-02-04T12:32:15.446Z",
        "max_reward": 0.001,
        "min_reward": 0.001,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "filterKey": "all",
        "history": Object {
          "action": "PUSH",
          "length": 46,
        },
        "proposition": Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;

exports[`Campaign snapshot renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <InjectIntl(withRouter(Campaign))
      filterKey="all"
      history={
        Object {
          "action": "PUSH",
          "length": 46,
        }
      }
      proposition={
        Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        }
      }
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "filterKey": "all",
      "history": Object {
        "action": "PUSH",
        "length": 46,
      },
      "proposition": Object {
        "count": 1,
        "distance": null,
        "last_created": "2020-02-04T12:32:15.446Z",
        "max_reward": 0.001,
        "min_reward": 0.001,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "filterKey": "all",
        "history": Object {
          "action": "PUSH",
          "length": 46,
        },
        "proposition": Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;

exports[`aa renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <InjectIntl(withRouter(Campaign))
      filterKey="all"
      history={
        Object {
          "action": "PUSH",
          "length": 46,
        }
      }
      proposition={
        Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        }
      }
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "filterKey": "all",
      "history": Object {
        "action": "PUSH",
        "length": 46,
      },
      "proposition": Object {
        "count": 1,
        "distance": null,
        "last_created": "2020-02-04T12:32:15.446Z",
        "max_reward": 0.001,
        "min_reward": 0.001,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "filterKey": "all",
        "history": Object {
          "action": "PUSH",
          "length": 46,
        },
        "proposition": Object {
          "count": 1,
          "distance": null,
          "last_created": "2020-02-04T12:32:15.446Z",
          "max_reward": 0.001,
          "min_reward": 0.001,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;
